### PR TITLE
feat: better logging

### PR DESCRIPTION
**Pull request by G0maa**
1. Write logs to file groups (db, graphql, error).
2. Rotate logs (every 14 days).
3. Log GraphQL requests (specifics might change on production).

This will be needed by #57.

### CODE REVIEW

File: apps/api/src/utils/winston.config.ts
Comment (bassiounix) [inline apps/api/src/utils/winston.config.ts:1]: This is not the place of configs. A separate config module is required.

File: apps/api/src/utils/graphql.logger.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {
+    ApolloServerPlugin,
+    BaseContext,
+    GraphQLRequestContext,
+    GraphQLRequestContextWillSendResponse,
+    GraphQLRequestListener,
```
Comment (bassiounix) [inline apps/api/src/utils/graphql.logger.ts:6]: Unused import
```suggestion
    Unused import `GraphQLRequestListener`
```

File: apps/api/src/utils/graphql.logger.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+} from "@apollo/server";
+import { Plugin } from "@nestjs/apollo";
+import { Logger } from "@nestjs/common";
+import { performance } from "perf_hooks";
+
+@Plugin()
+export class ApolloLogger implements ApolloServerPlugin {
+    private readonly logger = new Logger("GraphQL");
+
+    async requestDidStart(requestContext: GraphQLRequestContext<BaseContext>) {
```
Comment (bassiounix) [inline apps/api/src/utils/graphql.logger.ts:16]: Mark full access specifier as the `logger`
```suggestion
    public async requestDidStart(requestContext: GraphQLRequestContext<BaseContext>) {
```

File: apps/api/src/drizzle/transaction.interceptor.ts [hunk 3]
```diff
@@ -26,7 +30,10 @@ export class TransactionInterceptor implements NestInterceptor {
         const gqlContext = GqlExecutionContext.create(context);
         const ctx = gqlContext.getContext();
 
-        const db = drizzle(this.pool, { schema });
+        const db = drizzle(this.pool, {
```
Comment (bassiounix) [inline apps/api/src/drizzle/transaction.interceptor.ts:33]: Why the db is not injected?

**Review by bassiounix — APPROVED** [review]
LGTM. However, I didn't try to run logs locally!

File: apps/api/src/utils/winston.config.ts
Comment (G0maa) [reply apps/api/src/utils/winston.config.ts:1]: @bassiounix This is a feature request, we can create a separate issue for this.

The intention of this PR is to add logging, not create a config module.

File: apps/api/src/drizzle/transaction.interceptor.ts (hunk 3 above)
Comment (G0maa) [reply apps/api/src/drizzle/transaction.interceptor.ts:33]: Ya 3ammena, separate issue wallah :sob: See #2

**Pull request merged**